process.sh

```diff
@@ -1,37 +1,66 @@
 #!/bin/bash
 
-TIMES_DIR=/home/ncameron/times
-SCRIPTS_DIR=/home/ncameron/times-scripts
+if [ -z "$TIMES_DIR" ]; then
+    TIMES_DIR=/home/ncameron/times
+fi
+if [ -z "$SCRIPTS_DIR" ]; then
+    SCRIPTS_DIR=/home/ncameron/times-scripts
+fi
+
+echo TIMES_DIR=$TIMES_DIR
+echo SCRIPTS_DIR=$SCRIPTS_DIR
 
 START=$(pwd)
 export CARGO_RUSTC_OPTS="-Ztime-passes -Zinput-stats"
 export PATH=$RUSTC_DIR/bin:$PATH
 
-for dir in *; do
+# Check if user provided list of directories;
+# else process them all.
+if [ "$1" != "" ]; then
+    DIRS="$@"
+else
+    DIRS="*"
+fi    
+
+for dir in $DIRS; do
     if [[ -d $dir ]]; then
         echo "Processing $dir"
 
+        cd $START/$dir
+        PATCHES=($(make patches))
+        if [ ! "${PATCHES[*]}" ]; then
+            PATCHES=('')
+        fi
+
+        echo Patches: ${PATCHES[*]}
+
         for i in 0 1 2 3 4 5
         do
-            cd $RUST_DIR
-            git show HEAD -s >$TIMES_DIR/raw/$dir--$DATE--$i.log
-            cd $START/$dir
-            echo "rustc: ./$dir" >>$TIMES_DIR/raw/$dir--$DATE--$i.log
-            make >>$TIMES_DIR/raw/$dir--$DATE--$i.log
-            echo "done" >>$TIMES_DIR/raw/$dir--$DATE--$i.log
-
+            for PATCH in "${PATCHES[@]}"; do
+                cd $RUST_DIR
+                git show HEAD -s >$TIMES_DIR/raw/$dir$PATCH--$DATE--$i.log
+                cd $START/$dir
+                echo "rustc: ./$dir" >>$TIMES_DIR/raw/$dir$PATCH--$DATE--$i.log
```

Comment:
I think I need to actually to change this to `rustc: ./$dir$PATH`, but that will presumably require updating process.py as well